.github/workflows/memcheck.yml

```diff
@@ -101,6 +102,21 @@ jobs:
           SN_LOG: "all"
         timeout-minutes: 25
 
+      # Uploading same file using different client shall not incur any payment neither uploads
+      - name: Start a different client to upload the same file
+        run: |
+          mv $CLIENT_DATA_PATH $NODE_DATA_PATH/client_1
+          cargo run --bin faucet --release -- --log-output-dest=data-dir send 5000000 $(cargo run --bin safe --release -- --log-output-dest=data-dir wallet address | tail -n 1) > initial_balance_from_faucet_1.txt
+          cat initial_balance_from_faucet_1.txt
+          cat initial_balance_from_faucet_1.txt | tail -n 1 > transfer_hex
+          cat transfer_hex
+          cargo run --bin safe --release -- --log-output-dest=data-dir wallet receive --file transfer_hex
+          cargo run --bin safe --release -- --log-output-dest=data-dir files upload "./the-test-data.zip" --show-holders > second_upload.txt
+          cat second_upload.txt
```

Comment:
can we make this a full test and check the output here has no spend? (or am I missing that?)

Reply:
added an `rg` check to confirm the balance of wallet remain unchanged